From 1f976a6d03092c04c08ea799839b8f590c7d4932 Mon Sep 17 00:00:00 2001
From: Derek Chen <chenche@codeaurora.org>
Date: Wed, 5 Feb 2020 11:14:56 -0500
Subject: [PATCH] ASoC: msm: support dual Quat MI2S over TDM interface

* Modify number of channels and slot mappings for QUAT_TDM_RX_0
  and QUAT_TDM_RX_1
* Modify number of slots and slot width for QUAT_TDM interface

Suggested-by: Tahir Dawson <dawson@qti.qualcomm.com>
Signed-off-by: Derek Chen <chenche@codeaurora.org>
Change-Id: I95689624f95a3328fd5853936f2179d9b2fc3a41
---
 asoc/sa8155.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/asoc/sa8155.c b/asoc/sa8155.c
index a3af4ba..af4cf19 100644
--- a/asoc/sa8155.c
+++ b/asoc/sa8155.c
@@ -212,8 +212,8 @@ static struct dev_config tdm_rx_cfg[TDM_INTERFACE_MAX][TDM_PORT_MAX] = {
 		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 1}, /* RX_7 */
 	},
 	{ /* QUAT TDM */
-		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 8}, /* RX_0 */
-		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 8}, /* RX_1 */
+		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 2}, /* RX_0 */
+		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 2}, /* RX_1 */
 		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 1}, /* RX_2 */
 		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 1}, /* RX_3 */
 		{SAMPLING_RATE_48KHZ, SNDRV_PCM_FORMAT_S16_LE, 1}, /* RX_4 */
@@ -357,7 +357,7 @@ static struct tdm_slot_cfg tdm_slot[TDM_INTERFACE_MAX] = {
 	/* TERT TDM */
 	{32, 8},
 	/* QUAT TDM */
-	{32, 16},
+	{16, 2},
 	/* QUIN TDM */
 	{32, 16}
 };
@@ -415,8 +415,8 @@ static unsigned int tdm_rx_slot_offset
 		{0xFFFF}, /* not used */
 	},
 	{/* QUAT TDM */
-		{0, 8, 16, 24, 32, 40, 48, 56, 0xFFFF}, /*8 CH SPKR*/
-		{4, 12, 20, 28, 36, 44, 52, 60, 0xFFFF}, /*8 CH SPKR*/
+		{0, 2, 0xFFFF}, /*8 CH SPKR*/
+		{4, 6, 0xFFFF}, /*8 CH SPKR*/
 		{0xFFFF}, /* not used */
 		{0xFFFF}, /* not used */
 		{0xFFFF}, /* not used */
-- 
2.7.4

